### PR TITLE
Simulator Modification #Bug_Fix

### PR DESCRIPTION
**Pull request by syedomarazam**
Fixed issues for Task#D raised by Cristina Ruiz Martin

Created a .hpp and .cpp file for new function (transform_output()) to write formatted data to output file. 

Modifications are in below files :
1) See include/transform_output/transform_output.hpp file
2) See src/transform_output/transform_output.cpp file
3) makefile
4) README.md

old pull request : https://github.com/mohamedali5/AlternateBitProtocolSimulator/pull/4

### CODE REVIEW

**Review by p-sky — APPROVED** [review]

File: src/transform_output/transform_output.cpp [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+						in = token_4;
+						port_value = in.substr( 1 , ( in.find_first_of('}', 0) - 1) );
+	        			
+	        			/**
+	        			 *Getting our data and writing it to output file.
+	        			 * if time is not zero then we write the line to output file.
+	        			 */ 
+	        			if (outputfile.is_open()){
+
+	        			//	if(time!="00:00:00:000"){
```
Comment (shubhamagrawal6629) [inline src/transform_output/transform_output.cpp]: please remove commented code

File: src/transform_output/transform_output.cpp [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+ */
+std::string& rtrim(std::string& str, const std::string& chars = "\t\n\v\f\r ")
+{
+    str.erase(str.find_last_not_of(chars) + 1);
+    return str;
+}
+
+void transform_output(const char *ip_file_name, const char *op_file_name) {
+	
+	/**
```
Comment (shubhamagrawal6629) [inline src/transform_output/transform_output.cpp:45]: If you are considering this for the document then its ok 
Else you can use /* coment*/ for commenting variables

File: src/transform_output/transform_output.cpp [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+					
+			        token_3 = strtok(NULL, ","); 
+			    } 
+			    
+			}
+			
+		}
+	}
+	
+	//Close the file if still open.
```
Comment (shubhamagrawal6629) [inline src/transform_output/transform_output.cpp]: Use same as mentioned above

**Review by shubhamagrawal6629 — CHANGES_REQUESTED** [review]

**Comment by shubhamagrawal6629** [comment]
As mentioned in Part E comments here also if i delete your previous output file then your program does not work
As discussed in the class you need to generate both files.

Also as mentioned by prof in part E your gitignore is not working

So try bin/ and build / and try to push then

**Comment by syedomarazam** [comment]
@shubhamagrawal6629 : i did testing several times on Task#D. I have deleted all the output files before executing the .exe files. It generates as expected. I don't know how you are executing .exe files? I use the following commands in the Cygwin terminal. It works fine in all of my group members laptops.

./bin/top_model/ABP.exe data/top_model/input_abp_0.txt
./bin/test/SUBNET_TEST.exe
./bin/test/RECEIVER_TEST.exe 
./bin/test/SENDER_TEST.exe

File: src/transform_output/transform_output.cpp (hunk 2 above)
Comment (syedomarazam) [reply src/transform_output/transform_output.cpp:45]: yes.

File: src/transform_output/transform_output.cpp (hunk 3 above)
Comment (syedomarazam) [reply src/transform_output/transform_output.cpp]: fixed

File: src/transform_output/transform_output.cpp (hunk 1 above)
Comment (syedomarazam) [reply src/transform_output/transform_output.cpp]: fixed

**Comment by shubhamagrawal6629** [comment]
> ./bin/test/SUBNET_TEST.exe

ok my bad i was going inside bin folder and running 
Sorry for that

**Comment by shubhamagrawal6629** [comment]
Just gitignore remains that you can edit in part E
i will approve this one in a minute

**Review by shubhamagrawal6629 — APPROVED** [review]
I have checked all the files and the changes have been made as asked and the program runs properly and generates output properly

**Pull request merged**